src/main/kotlin/br/unipar/plano/domain/solicitacaoprocedimento/service/impl/ValidadorSolicitacaoProcedimento.kt

```diff
@@ -0,0 +1,37 @@
+package br.unipar.plano.domain.solicitacaoprocedimento.service.impl
+
+import br.unipar.plano.domain.procedimento.model.IdProcedimento
+import br.unipar.plano.domain.solicitacaoprocedimento.model.SolicitacaoProcedimento
+import br.unipar.plano.domain.solicitacaoprocedimento.model.StatusSolicitacaoProcedimento
+import br.unipar.plano.domain.solicitacaoprocedimento.service.SolicitacaoProcedimentoRepository
+import java.time.Duration
+
+class ValidadorSolicitacaoProcedimento {
```

Comment:
O validador pode ser um service que você injeta nos casos de uso, nesse caso o métodos não podem ser estáticos e você deve receber o repositório no construtor. Essa é uma boa prática pois assim você pode criar mocks dessas classes quando necessário para teste.

```suggestion
@Service
class ValidadorSolicitacaoProcedimento {
```